src/librustdoc/html/render/mod.rs

```diff
@@ -1552,34 +1551,27 @@ fn render_impl(
             );
         }
     }
-    let toggled = !impl_items.is_empty() || !default_impl_items.is_empty();
-    let open_details = |close_tags: &mut String, is_collapsed: bool| {
+    let on_trait_page = matches!(*parent.kind, clean::ItemKind::TraitItem(_));
+    let has_impl_items = !(impl_items.is_empty() && default_impl_items.is_empty());
+    let toggled = !on_trait_page && has_impl_items;
```

Comment:
Why do we want to check if we are on a trait page now? More like: why wasn't it an issue before?